core/platform/lf_STM32f4_support.c

```diff
@@ -0,0 +1,251 @@
+    TIM5->DIER |= TIM_DIER_CC1IE;
+    NVIC_EnableIRQ(TIM5_IRQn);
+
+    /* This is to make the Prescaler actually work
+     *  For some reason, the Prescaler only kicks in once the timer has reset once.
+     *  Thus, the current solution is to manually ret the value to a really large
+     *  and force it to reset once. Its really jank but its the only way I could
+     *  find to make it work
+     */
+    TIM5->CNT = 0xFFFFFFFE;
+}
+
+/**
+ * ISR for handling timer overflow -> We increment the upper timer
+ */
+void TIM5_IRQHandler(void) {
+    if (TIM5->SR & (1 << 1)) {
+        TIM5->SR &= ~(1 << 1);
+        _lf_time_us_high += 1;
+    }
+}
+
+/**
+ * Write the time since boot into time variable
+ */
+int _lf_clock_now(instant_t *t)
+{
+    // Timer is cooked
+    if (!t) {
+        return -1;
+    }
+    // Get the current microseconds from TIM5
+    uint32_t _lf_time_us_low = TIM5->CNT;
+
+    // Combine upper and lower timers (Yoinked from lf_nrf52 support)
+    uint64_t now_us = COMBINE_HI_LO((_lf_time_us_high - 1), _lf_time_us_low);
+    *t = ((instant_t)now_us) * 1000;
+    return 0;
+}
+
+/**
+ * Make the STM32 go honk shoo mimimi for set nanoseconds
+ * I essentially stole this from the lf_nrf52 support
+ */
+int lf_sleep(interval_t sleep_duration) {
+    instant_t target_time;
+    instant_t current_time;
+
+    _lf_clock_now(&current_time);
+    target_time = current_time + sleep_duration;
+    while (current_time <= target_time)
+        _lf_clock_now(&current_time);
+
+    return 0;
+}
+
+/**
+ * Make the STM32 go honk shoo honk shoo for set nanoseconds
+ * This one uses a do-while loop. :)
+ * I essentially stole this from the lf_nrf52 support
+ */
+static void lf_busy_wait_until(instant_t wakeup_time) {
+    instant_t now;
+    do {
+        _lf_clock_now(&now);
+    } while (now < wakeup_time);
+}
+
+// I am pretty sure this function doesnt work
+//      Ill try to fix it once i know what the fuck its supposed to do, LOL
+/*  sleep until wakeup time
+    But, wake up if there is an async event
+
+*/
+int _lf_interruptable_sleep_until_locked(environment_t *env, instant_t wakeup_time) {
+    // Get the current time and sleep time
+    instant_t now;
+    _lf_clock_now(&now);
+    interval_t duration = wakeup_time - now;
+
+    // Edge case handling for super small duration
+    if (duration <= 0) {
+        return 0;
+    } else if (duration < 10) {
+        lf_busy_wait_until(wakeup_time);
+        return 0;
+    }
+
+    // Enable interrupts and prepare to wait
+    _lf_async_event = false;
+    lf_enable_interrupts_nested();
+
+    do {
+        _lf_clock_now(&now);
+
+        // Exit when the timer is up or there is an exception
+    } while (!_lf_async_event && (now < wakeup_time));
+
+    // Disable interrupts again on exit
+    lf_disable_interrupts_nested();
+
+    if (!_lf_async_event) {
+        return 0;
+    } else {
+        LF_PRINT_DEBUG(" *The STM32 rises from sleep* \n");
```

Comment:
Are there timing implications of `LF_PRINT_DEBUG` that should be considered here? Is this message necessary?